File: simpleMediaPlayer/app/src/main/java/com/daeseong/simplemediaplayer/PlayerTimer.java
package com.daeseong.simplemediaplayer;

import android.os.Handler;
import android.os.Looper;

public class PlayerTimer {

    private static final int DEFAULT_INTERVAL_MILLIS = 1000;

    private boolean isUpdate = false;
    private Callback callback;
    private long startTimeMillis;
    private Handler handler;

    public PlayerTimer() {
        handler = new Handler(Looper.getMainLooper());
    }

    private void init() {
        startTimeMillis = System.currentTimeMillis();
    }

    public void start() {
        init();
        isUpdate = true;
        handler.post(timerRunnable);
    }

    public void stop() {
        isUpdate = false;
        handler.removeCallbacksAndMessages(null);
    }

    public void setCallback(Callback callback) {
        this.callback = callback;
    }

    private final Runnable timerRunnable = new Runnable() {
        @Override
        public void run() {
            if (isUpdate) {
                callback.onTick(System.currentTimeMillis() - startTimeMillis);
                handler.postDelayed(this, DEFAULT_INTERVAL_MILLIS);
            }
        }
    };

    public interface Callback {
        void onTick(long timeMillis);
    }
}

File: mediaplayer_test(kotlin)/app/src/main/java/com/daeseong/mediaplayer_test/PlayerTimer.kt
package com.daeseong.mediaplayer_test

import android.os.Handler
import android.os.Looper

class PlayerTimer {
    private val DEFAULT_INTERVAL_MILLIS = 1000L
    private var isUpdate = false
    private var callback: Callback? = null
    private var startTimeMillis: Long = 0
    private val handler: Handler = Handler(Looper.getMainLooper())

    private fun init() {
        startTimeMillis = System.currentTimeMillis()
    }

    fun start() {
        init()
        isUpdate = true
        handler.post(timerRunnable)
    }

    fun stop() {
        isUpdate = false
        handler.removeCallbacksAndMessages(null)
    }

    fun setCallback(callback: Callback) {
        this.callback = callback
    }

    private val timerRunnable: Runnable = object : Runnable {
        override fun run() {
            if (isUpdate) {
                callback?.onTick(System.currentTimeMillis() - startTimeMillis)
                handler.postDelayed(this, DEFAULT_INTERVAL_MILLIS)
            }
        }
    }

    interface Callback {
        fun onTick(timeMillis: Long)
    }
}